src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -86,32 +86,29 @@ public async Task<ActionResult<DataElement>> Delete(int instanceOwnerPartyId, Gu
                 return NotFound("Provided instanceId is unknown to storage service");
             }
 
-            using (OrgDataContext context = _dataRepository.GetOrgDataContext(instance.Org))
-            {
-                DataElement dataElement = await _dataRepository.Read(instanceGuid, dataGuid);
+            DataElement dataElement = await _dataRepository.Read(instanceGuid, dataGuid);
 
-                if (dataElement == null)
-                {
-                    return NotFound("Provided dataGuid is unknown to storage service");
-                }
+            if (dataElement == null)
+            {
+                return NotFound("Provided dataGuid is unknown to storage service");
+            }
 
-                try
-                {
-                    string storageFileName = DataElementHelper.DataFileName(instance.AppId, instanceGuid.ToString(), dataGuid.ToString());
-                    bool result = await _dataRepository.DeleteDataInStorage(storageFileName);
+            try
+            {
+                string storageFileName = DataElementHelper.DataFileName(instance.AppId, instanceGuid.ToString(), dataGuid.ToString());
+                _ = await _dataRepository.DeleteDataInStorage(instance.Org, storageFileName);
 
-                    await _dataRepository.Delete(dataElement);
+                await _dataRepository.Delete(dataElement);
 
-                    await DispatchEvent(InstanceEventType.Deleted.ToString(), instance, dataElement);
+                await DispatchEvent(InstanceEventType.Deleted.ToString(), instance, dataElement);
 
-                    return Ok(dataElement);
-                }
-                catch (Exception deleteException)
-                {
-                    _logger.LogError($"Unable to delete data element {dataGuid} due to {deleteException}");
+                return Ok(dataElement);
+            }
+            catch (Exception deleteException)
+            {
+                _logger.LogError($"Unable to delete data element {dataGuid} due to {deleteException}");
 
-                    return StatusCode(500, $"Unable to delete data element {dataGuid} due to {deleteException.Message}");
-                }
+                return StatusCode(500, $"Unable to delete data element {dataGuid} due to {deleteException.Message}");
```

Comment:
Who has access to this API? Could it be a problem to return potential stack info here?

Reply:
The application will use this API. As attachments are saved/deleted instantly based on user interaction with app frontend. Currently the error would be hidden by app backend anyways, so we should make create a new issue for this case or add it as a point in  #2304

Reply:
We need to consider standardizing error handling in all platform application so they follow the same pattern. Explicit return of status code 500 can't be right.
With that being said I haven't looked at these things in this change. It is big enough as it is.